lib/github/ldap/user_search/active_directory.rb

```diff
@@ -0,0 +1,38 @@
+module GitHub
+  class Ldap
+    module UserSearch
+      class ActiveDirectory < Default
+
+        # Public - Overridden from base class to set the base to "", and use the
+        # Global Catalog to perform the user search.
+        def search(search_options)
+          # when doing a global search for a user's DN, set the search base to blank
+          options[:base] = ""
```

Comment:
Overriding the `@options` value here feels wrong. Perhaps we can override `options` to include this value by default in this strategy?